pkg/pillar/nireconciler/genericitems/registry.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2023 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package genericitems
+
+import (
+	"github.com/lf-edge/eve/libs/reconciler"
+	"github.com/lf-edge/eve/pkg/pillar/base"
+	"github.com/sirupsen/logrus"
+)
+
+// RegisterItems : register all configurators implemented by this package.
+func RegisterItems(log *base.LogObject, logger *logrus.Logger,
```

Comment:
Why do we have to register all configurators?

Reply:
Because they are all used :)

Reply:
I understand they are used, but I  don't get the registration mechanism and why do we need it.

Reply:
It is mentioned here: https://github.com/lf-edge/eve/tree/master/libs/reconciler#configurator-registry
Registry creates mapping between an Item and a Configurator, so that Reconciler knows which Configurator to use to create/modify/delete a particular Item.